docs/interactions/Slash_Commands.md

```diff
@@ -0,0 +1,837 @@
+                        {
+                            "name": "user",
+                            "description": "The user to edit",
+                            "type": 6,
+                            "required": true
+                        },
+                        {
+                            "name": "channel",
+                            "description": "The channel permissions to edit",
+                            "type": 7,
+                            "required": false
+                        }
+                    ]
+                }
+            ]
+        },
+        {
+            "name": "role",
+            "description": "Get or edit permissions for a role",
+            "type": 2,
+            "options": [
+                {
+                    "name": "get",
+                    "description": "Get permissions for a role",
+                    "type": 1,
+                    "options": [
+                        {
+                            "name": "role",
+                            "description": "The role to get",
+                            "type": 8, // 8 is type ROLE
+                            "required": true
+                        },
+                        {
+                            "name": "channel",
+                            "description": "The channel permissions to get",
+                            "type": 7,
+                            "required": false
+                        }
+                    ]
+                },
+                {
+                    "name": "edit",
+                    "description": "Edit permissions for a role",
+                    "type": 1,
+                    "options": [
+                        {
+                            "name": "role",
+                            "description": "The role to edit",
+                            "type": 8,
+                            "required": true
+                        },
+                        {
+                            "name": "channel",
+                            "description": "The channel permissions to edit",
+                            "type": 7,
+                            "required": false
+                        }
+                    ]
+                }
+            ]
+        }
+    ]
+}
+```
+
+And, done! The JSON looks a bit complicated, but what we've ended up with is a single command that can be scoped to multiple actions, and then further scoped to a particular resource, and then even _further_ scope with optional arguments. Here's what it looks like all put together.
+
+![A command with grouped subcommands and parameters. It says /permissions user get with arguments for a user and a channel.](command-with-groups-subcommands-parameters.png)
+
+## Endpoints
+
+> info
+> For authoriation, all endpoints take either a bot token or client credentials token for your application
+
+## Get Global Application Commands % GET /applications/{application.id}/commands
```

Comment:
`{application.id}` should probably link to the application object; would be good to merge #1838 first so its in a better place

also can application data in READY be documented? would be useful for libs/old bots implementing this feature

Reply:
Funny enough I think we only document the application object as a response to the GET endpoint. Probably need to refactor that a bit

Reply:
the interaction response & followup message endpoints still arent linked